<a href="https://colab.research.google.com/github/Nikolai0Huarcaya/C8280/blob/main/Actividad.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Pregunta 1.- Mire en la parte superior e inferior de data.tsv entregado en esta actividad. Piensa en cómo se estructuran los datos. ¿Qué crees que significa cada columna? ¿Qué crees que significa cada fila?**
*   Respuesta 1 : Primero mencionaremos con respecto a las columnas, que son los datos que quedan registrados en la bas de datos una vez que son ingresados como se puede mostrar en los códigos de parte de abajo.
*   Respuesta 2: Por otro lado las filas son las carácterísticas, como son el número total de datos, se podría ser que son la cantidad pedida por cada dato ingresado, obviamente los datos ingresados y trambién la descripción de ellos tendra un platillo o tabla que se demuestra exprese.




In [41]:
import pandas as pd
url='https://raw.githubusercontent.com/Nikolai0Huarcaya/C8280/main/Actividad1/data.tsv'

In [42]:
Datos = pd.read_table(url)
Datos

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

**A continuación se mostrara cuantas columnas y filas tiene este data.tsv**

In [28]:
Datos.shape

(4622, 5)

In [30]:
Datos.columns

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

In [33]:
columnas = Datos.columns
columnas

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

In [36]:
type(columnas)

pandas.core.indexes.base.Index

**Aquí con este código podemos observar el indice, el número de columnas del dataset más especificado**


In [ ]:
Datos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


**En esta sección cambiamos el archivo tsv a csv que seguidamente se subio al directorio, pero como se puede obsevar no hay cambios que afecten a la ejecución del código**

In [49]:
import pandas as pd

In [53]:
url="https://raw.githubusercontent.com/Nikolai0Huarcaya/C8280/main/Actividad1/data.csv"

In [54]:
datos=pd.read_csv(url)
datos

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

**2.- ¿Cuántos pedidos parece haber?**




In [59]:
pedidos = Datos["quantity"].sum()
print("pedidos:" , pedidos)

pedidos: 4972


Cómo podemos ver hay 4972 pedidos realizados en el data.


**3.-¿Cuántas líneas hay en este archivo?**



In [60]:
len(Datos)

4622

Con el siguiente código podemos saber que en el archivo se encuentran 4622 líneas.

**4.-¿Qué burrito es más popular, steak o chicken?**